app/src/main/java/org/schulcloud/mobile/models/file/FileRepository.kt

```diff
@@ -29,23 +29,13 @@ object FileRepository : Repository() {
 
     suspend fun syncDirectory(owner: String, parent: String?) {
         RequestJob.Data.with({ listDirectoryContents(owner, parent) }, {
-                beginGroup()
-                        .equalTo("isDirectory", false)
-                        .equalTo("owner", owner)
-                        .or()
-                        .notEqualTo("owner", owner)
-                        .endGroup()
-    }).run()
+            equalTo("isDirectory", false)
```

Comment:
This still looks off. If one folder is synced you delete all known directories?

Reply:
It's for the BreadcrumbsView and url: To compute the path of a file/directory I need the parent directories (like recursive up to the owner, which is either a course or the user), so I fetch all directories for the owner at once (probably faster than fetching them one by one). When fetching the directory contents for a parent directory I have to make sure these other directories are not deleted (otherwise they would be deleted, because they have a different parent), so I delete the obsolete directories only when I fetch the directories for the owner. Would be cleaner if I fetched the needed directories one by one, but that would result in more server requests which is slower.